lib/CabalToDhall.hs

```diff
@@ -621,35 +614,35 @@ licenseToDhall =
           Right ( Cabal.LGPL v ) ->
             license "LGPL" ( Dhall.embed ( maybeToDhall versionToDhall ) v )
           Right Cabal.BSD2 ->
-            license "BSD2" ( Expr.RecordLit mempty )
+            licenseNullary "BSD2"
           Right Cabal.BSD3 ->
-            license "BSD3" ( Expr.RecordLit mempty )
+            licenseNullary "BSD3"
           Right Cabal.BSD4 ->
-            license "BSD4" ( Expr.RecordLit mempty )
+            licenseNullary "BSD4"
           Right Cabal.MIT ->
-            license "MIT" ( Expr.RecordLit mempty )
+            licenseNullary "MIT"
           Right Cabal.ISC ->
-            license "ISC" ( Expr.RecordLit mempty )
+            licenseNullary "ISC"
           Right ( Cabal.MPL v ) ->
             license "MPL" ( Dhall.embed versionToDhall v )
           Right ( Cabal.Apache v ) ->
             license "Apache" ( Dhall.embed ( maybeToDhall versionToDhall ) v )
           Right Cabal.PublicDomain ->
-            license "PublicDomain" ( Expr.RecordLit mempty )
+            licenseNullary "PublicDomain"
           Right Cabal.AllRightsReserved ->
-            license "AllRightsReserved" ( Expr.RecordLit mempty )
+            licenseNullary "AllRightsReserved"
           -- Note: SPDX.NONE is what Cabal reports for a file without
           -- a 'license' field, even for pre-2.2 spec versions.
           Left SPDX.NONE ->
-            license "AllRightsReserved" ( Expr.RecordLit mempty )
+            licenseNullary "AllRightsReserved"
           Right Cabal.UnspecifiedLicense ->
-            license "Unspecified" ( Expr.RecordLit mempty )
+            licenseNullary "Unspecified"
           Right ( Cabal.UnknownLicense "UnspecifiedLicense" ) ->
-            license "Unspecified" ( Expr.RecordLit mempty )
+            licenseNullary "Unspecified"
           Right ( Cabal.UnknownLicense l ) ->
             license "Unspecified" ( Expr.TextLit (Expr.Chunks [] (StrictText.pack l)) )
```

Comment:
maybe it should be
```suggestion
             license "Unknown" ( Expr.TextLit (Expr.Chunks [] (StrictText.pack l)) )
```

Reply:
Plausible. I think it's worth keeping the meaningful changes in this PR to a minimum, though - if this code is confused, it's not as a result of the 1.23-related changes, and it can be dealt with separately.

Reply:
Agree. Maybe you can't write a cabal config in the actual version to reach that code path (so in that case maybe the union case could be removed from dhall-to-cabal)